pkg/canary/comparator/comparator.go

```diff
@@ -246,7 +247,10 @@ func (c *Comparator) Size() int {
 
 func (c *Comparator) run() {
 	t := time.NewTicker(c.pruneInterval)
-	mt := time.NewTicker(c.metricTestInterval)
+	// Use a random tick up to the interval for the first tick
+	firstMt := true
+	rand.Seed(time.Now().UnixNano())
+	mt := time.NewTicker(time.Duration(rand.Int63n(c.metricTestInterval.Nanoseconds())))
```

Comment:
I think the default for this is `1hr`, so we delay the first tick between 0-3600s?

Reply:
assuming I did this correctly that's the idea, the base unit of a Duration is nanoseconds so we just spread the start time of this test for every canary randomly over the interval